lms/djangoapps/courseware/date_summary.py

```diff
@@ -204,6 +205,9 @@ class VerifiedUpgradeDeadlineDate(DateSummary):
 
     @property
     def link(self):
+        ecommerce_service = EcommerceService()
+        if ecommerce_service.is_enabled(self.user):
```

Comment:
It could just be that it is late, but I'm not seeing 'user' defined in this class or the DateSummary parent -- where is it coming from?